src/client/game.cpp

```diff
@@ -2190,8 +2190,12 @@ void Game::processItemSelection(u16 *new_playeritem)
 
 	/* Item selection using mouse wheel
 	 */
+	Inventory *inventory = &player->inventory;
+	InventoryList *mainlist = inventory->getList("main");
+	if (mainlist == nullptr)
+		return;
 	*new_playeritem = player->getWieldIndex();
-	u16 max_item = MYMIN(PLAYER_INVENTORY_SIZE - 1,
+	u16 max_item = MYMIN((s32) mainlist->getSize() - 1,
 		    player->hud_hotbar_itemcount - 1);
```

Comment:
1. `Player` contains `inventory` and `hud_hotbar_itemcount`
2. This code is technically duplicated: game.cpp (here), hud.cpp, l_object.cpp

Hence it would make sense to add a dedicated function to `Player` that retrieves the correctly limited hotbar count.

Reply:
Done in 22e5e8b92